### ISSUE TITLE
Optional authentication for about page

### ISSUE THREAD

**Issue by oxr463** [buildbot/buildbot#5080]
The about page contains version numbers for the software that is running on a public buildbot instance. I would like to ensure that this information can only be accessed by authenticated users. This is a common finding for web application penetration testing, (See: https://security.stackexchange.com/q/4940/209989); however, the severity is rather low so that is why I think it should be an optional change for those who are being audited.

**Comment by tardyp** [comment]
This is part of security best practice to not disclose those kind of information. 
However it is not very efficient in our case.
The checksum of the javascript files are changing for every release of buildbot, so it will be easy to know which version of buildbot you are using, by comparing with all published release. You can also deobfuscate the javascript and look for unsecure patterns.

So this best practice is more for backend code and should be only a part of the security practices (most important is to upgrade your deps regularly)

**Comment by tardyp** [comment]
On the other hand, the open source licenses we are using are requiring us to disclose that we use them, so legally, we need to tell that to all users we are distributing the sw, including non authenticated..

**Comment by oxr463** [comment]
I agree with you completely, hence my suggestion that this be an optional feature. Security compliance can be a bit silly at times, but if the effort to accommodate is unsubstantial, then I see no reason not to.
